app/assets/javascripts/project_find_file.js.coffee

```diff
@@ -0,0 +1,121 @@
+class @ProjectFindFile
+  constructor: (@element, @options)->
+    @filePaths = {}
+    @inputElement = @element.find(".file-finder-input")
+
+    # init event
+    @initEvent()
+
+    # focus text input box
+    @inputElement.focus()
+
+    # load file list
+    @load(@options.url)
+
+  # init key event
+  initEvent: ->
+    @inputElement.off "keyup"
+    @inputElement.on "keyup", (event) =>
+      target = $(event.target)
+      value = target.val()
+
+      if value != target.data("oldValue")
+        target.data("oldValue", value)
+        @findFile()
+        @element.find('tr.tree-item').eq(0).addClass("selected").focus()
+
+    @element.find(".tree-content-holder .tree-table").on 'click', (e) ->
+      if (e.target.nodeName != "A")
+        path = $('.tree-item-file-name a', this).attr('href')
+        if path then location.href = path
+
+  # find file
+  findFile: ->
+    searchTxt = @inputElement.val()
+    result = if searchTxt then fuzzaldrinPlus.filter(@filePaths, searchTxt) else @filePaths

```

Comment:
When `searchTxt` equals `""`, it will still be true and we enter fuzzAldrin. Please check if `searchTxt.length > 0`